app/controllers/blogs_controller.rb

```diff
@@ -0,0 +1,74 @@
+class BlogsController < ApplicationController
+  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+
+  # GET /blogs
+  # GET /blogs.json
+  def index
+    @blogs = Blog.all
+  end
+
+  # GET /blogs/1
+  # GET /blogs/1.json
+  def show; end
+
+  # GET /blogs/new
+  def new
+    @blog = current_user.blogs.build
+  end
```

Comment:
- **_It should initialize an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**